src/NHibernate/Dialect/MsSql2012Dialect.cs

```diff
@@ -30,7 +30,7 @@ protected override string GetCreateSequenceString(string sequenceName, int initi
 
 		public override string GetDropSequenceString(string sequenceName)
 		{
-			string dropSequence = "IF EXISTS (select * from sys.sequences where name = N'{0}') DROP SEQUENCE {0}";
+			string dropSequence = "if object_id(N'{0}') is not null drop sequence {0}";
```

Comment:
This should read as `IF EXISTS (select * from {catalog}.sys.sequences where object_id =object_id(N'{0}')) ...`

Taking into the consideration catalog+default_catalog, schema+default_schema.